python-package/xgboost/spark/data.py

```diff
@@ -1,194 +1,85 @@
-# type: ignore
-"""Xgboost pyspark integration submodule for data related functions."""
-# pylint: disable=too-many-arguments
-from typing import Iterator
+from collections import defaultdict, namedtuple
+from typing import Any, Dict, Iterator, List, Optional, Sequence, Tuple
 
 import numpy as np
 import pandas as pd
 
+from xgboost.compat import concat
 from xgboost import DMatrix
 
 
-def _prepare_train_val_data(
-    data_iterator, has_weight, has_validation, has_fit_base_margin
-):
-    def gen_data_pdf():
-        for pdf in data_iterator:
-            yield pdf
+def stack_series(df: pd.Series) -> np.ndarray:
+    array = df.to_numpy(copy=False)
+    if array.ndim == 1:
+        array = array.reshape(array.shape[0], 1)
+    array = np.stack(array[:, 0])
+    return array
```

Comment:
Do we need `reshape` here ?

e.g.

```
>>> p1 = pd.DataFrame({'a':[np.array([3,4]), np.array([4,5])]})
>>> p1
        a
0  [3, 4]
1  [4, 5]
>>> p1.a
>>> np.stack(p1.a.to_numpy())
array([[3, 4],
            [4, 5]])
```
I don't use `reshape` and directly stack series values , also got correct result.